addons/osio-addon/src/main/java/io/fabric8/launcher/osio/enrichers/GitUserGoEnricher.java

```diff
@@ -0,0 +1,98 @@
+
+    private static final List<String> VALID_EXTENSIONS = asList("go");
+
+    private static final List<String> VALID_FILENAMES = asList("Makefile", "assemble", "environment");
+
+    private final GitService gitService;
+
+    @Inject
+    public GitUserGoEnricher(GitService gitService) {
+        this.gitService = gitService;
+    }
+
+    @Override
+    public void accept(Projectile projectile) {
+        if (!(projectile instanceof OsioLaunchProjectile)) {
+            // If projectile isn't an OsioLaunchProjectile, skip it
+            return;
+        }
+        OsioLaunchProjectile proj = (OsioLaunchProjectile) projectile;
+        // If no organization specified, use the owner location instead
+        final String gitOrganization = proj.getGitOrganization() == null ?
+                gitService.getLoggedUser().getLogin() :
+                proj.getGitOrganization();
+        final String gitRepositoryName = proj.getGitRepositoryName();
+        String[] split = proj.getBooster().getGitRepo().split("/");
+        final String boosterGitOrganization = split[split.length - 2];
+        final String boosterGitRepository = split[split.length - 1];
+        try {
+            Files.walkFileTree(proj.getProjectLocation(), new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    String fileName = file.getFileName().toString();
+                    if (VALID_FILENAMES.contains(fileName) ||
+                            VALID_EXTENSIONS.contains(FilenameUtils.getExtension(fileName))) {
+                        customizeFile(file, gitOrganization, gitRepositoryName, boosterGitOrganization, boosterGitRepository);
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        } catch (IOException e) {
+            log.log(Level.SEVERE, "Error while walking files in repository", e);
+        }
+    }
+
+    /**
+     * Customizes the import section of .go files as well as the makefile and
+     * environment file to find and replace the git organization value.
+     */
+    private void customizeFile(Path path, String newGitOrg, String newGitRepo, String oldGitOrg, String oldGitRepo) throws IOException {
+        String contents = new String(Files.readAllBytes(path));
+        int importStartIdx = contents.indexOf(IMPORT_START);
+        if (importStartIdx < 0) {
+            String newContents = contents.replaceAll(oldGitOrg, newGitOrg).replaceAll(oldGitRepo, newGitRepo);
+            Files.write(path, newContents.getBytes());
+        } else {
+            int importEndIdx = contents.indexOf(IMPORT_END, importStartIdx);
+            String importContents = contents.substring(importStartIdx + IMPORT_START.length(), importEndIdx);
+            String newImportContents = importContents.replaceAll(oldGitOrg, newGitOrg).replaceAll(oldGitRepo, newGitRepo);
```

Comment:
I just realized that the replace should happen using the `org/repo` pattern, otherwise it will replace unwanted content

Reply:
done